autorest/azure/token.go

```diff
@@ -168,12 +191,24 @@ func NewServicePrincipalTokenWithSecret(id string, tenantID string, resource str
 		refreshWithin: defaultRefresh,
 		sender:        &http.Client{},
 	}
-	return spt, nil
+	return spt
+}
+
+// NewServicePrincipalTokenFromManualToken creates a ServicePrincipalToken using the supplied token
+func NewServicePrincipalTokenFromManualToken(id string, tenantID string, resource string, token Token) *ServicePrincipalToken {
```

Comment:
I think this method can take the `refreshCallback` value and store it internally. I'm not sure the `SetRefreshCallback` setter will be called twice in any scenario. Also by making it settable only in the constructor you make it concurrency-safe (in other words, Go assignments are not atomic).